src/encoder.rs

```diff
@@ -159,12 +161,22 @@ impl<InstructionsType: Instructions> Encoder<InstructionsType> {
         };
     }
 
+    fn push_path_condition(&mut self, cond: SMTVariable) {
+        self.path_conditions.push(cond);
+    }
+
+    fn pop_path_condition(&mut self) {
+        self.path_conditions.pop();
+    }
+
     fn encode_if(&mut self, expr: &yul::If) {
         let cond = self.encode_expression(&expr.condition);
         assert!(cond.len() == 1);
         let prev_ssa = self.ssa_tracker.copy_current_ssa();
 
+        self.push_path_condition(cond[0].clone());
```

Comment:
Ideally, we would combine this with joining the branches

Reply:
yea makes sense. The reason I kept is a draft is exactly cus it's not implemented for switch/for

Reply:
not sure we can actually, cus this is done wrapping the `encode_block` call, whereas `join_branches` happens later on